src/classes/dice/DiceRoller.ts

```diff
@@ -206,41 +206,57 @@ class DiceRoller {
     } else {
       let total = 0
       const rawRolls: number[] = []
-      const lowRolls: number[] = []
+      const rollClass: string[] = []
       let okRerolls = 0
       let staticBonus = 0
 
       staticBonus = parsedRoll.modifier
       total = staticBonus
 
       parsedRoll.dice.forEach(dieSet => {
-        if (critical) {
-          let x, y
-          x = DiceRoller.rollDieSet(dieSet, overkill)
-          y = DiceRoller.rollDieSet(dieSet, overkill)
-          if (x.result > y.result) {
-            rawRolls.push(...x.rolls)
-            lowRolls.push(...y.rolls)
-          }
-          else {
-            rawRolls.push(...y.rolls)
-            lowRolls.push(...x.rolls)
-          }
-          okRerolls += x.result > y.result ? x.rerolls : y.rerolls
-          total += x.result > y.result ? x.result : y.result
-        } else {
-          let x
-          x = DiceRoller.rollDieSet(dieSet, overkill)
-          rawRolls.push(...x.rolls)
-          okRerolls += x.rerolls
-          total += x.result
-        }
+        let x
+        x = DiceRoller.rollDieSet(dieSet, overkill, critical)
+        rawRolls.push(...x.rolls)
+        rollClass.push(...this.classifyDamageRolls(dieSet,x.rolls,overkill))
+        okRerolls += x.rerolls
+        total += x.result
       })
 
-      return new DamageRollResult(diceString, total, rawRolls, lowRolls, staticBonus, okRerolls, false)
+      return new DamageRollResult(diceString, total, rawRolls, rollClass, staticBonus, okRerolls, false)
     }
   }
 
+  public static classifyDamageRolls(dieSet: DieSet, rolls: number[], overkill?: boolean): string[] {
+    if (dieSet.quantity <= 0 || rolls.length <= 0) return []
+    
+    const rollClass: string[] = []
+    //Set initial 'overkill' and 'low' values
+    rolls.forEach((value, index) => {
+      if (overkill && (value === 1)) {
+        rollClass.push('overkill')
+      } else {
+        rollClass.push('low')
+      }
+    })
+
+    //Find high values and mark them
+    for (let x = 0; x < dieSet.quantity; x++) {
+      let highValue = 0
+      let highIndex = 0;
+      rolls.forEach((value, index) => {
+        if(value > highValue && rollClass[index] != 'high') {
+          highValue = value
+          highIndex = index
+        }
+      })
+      if(highValue > 0) {
+        rollClass[highIndex] = 'high'
+      }
+    }
```

Comment:
This loop too, I'm not very used to thinking or using functional programming concepts, so if there is a better / more consistent way to find and mark the highest unique `n` values, please let me know and I'll happily change it.